Add guide on how to install 'test' versions (#232)
contao/docs: docs/manual/guides/install-test-versions.en.md

---
title: "Installing test versions"
description: "Guide on how to install development versions and release candidates of Contao."
weight: 20
---

The development team of Contao adheres to a specific [release plan][releasePlan]
when developing new versions of Contao (and maintaining current versions). Before

> each planned major and minor version release there will a specific timeframe for

Reviewer: ```suggestion
each planned major and minor version release there will be a specific timeframe for
```

developing this new version. New features will be added until a deadline is reached,
after which the first release candidates will be published. If no further issues
are found during testing of the developer version and the release candidates, the
stable version of the new version will be released.

Even though Contao uses unit testing, unforeseen issues can always occur after implementing

> new features or changing existing ones. Thus testing still needs to be done by actual 

Reviewer: ```suggestion
new features or changing existing ones. Thus, testing still needs to be done by actual 
```

users. Since Contao is an open source software it can benefit from community engagement
for such tests.

The following explains how you can participate in testing, by installing either
a release candidate of Contao during its release candidate phase, or even a development
version of Contao.

When updating packages via Composer (either directly via the command line or via
the Contao Manager), usually only _stable_ packages will be installed, at least

> with the default setup of the _Contao Managed Edition_ for example. Therefore we

Reviewer: ```suggestion
with the default setup of the _Contao Managed Edition_ for example. Therefore, we
```

> one needs to make a few adjustments to the `composer.json` of your project, in order

Reviewer: ```suggestion
only need to make a few adjustments to the `composer.json` of your project, in order
```

        "contao/news-bundle": "^4.9@RC",
        "contao/newsletter-bundle": "^4.9@RC"
    },
    "conflict": {
        "contao-components/installer": "<1.3"
    },
    "extra": {
        "contao-component-dir": "assets",
        "symfony": {
            "require": "^4.2"
        }
    },
    "scripts": {
        "post-install-cmd": [
            "Contao\\ManagerBundle\\Composer\\ScriptHandler::initializeApplication"
        ],
        "post-update-cmd": [
            "Contao\\ManagerBundle\\Composer\\ScriptHandler::initializeApplication"
        ]
    }
}
```

Note that once Contao `4.9` has stable releases (e.g. `4.9.0`, `4.9.1` etc.), those 
will be installed when running a package update.

## Installing Developer Versions

During the release candidate and development phase of a Contao version, you can
also install the development version for testing purposes instead of just the already
defined release candidates. This way you can test the latest changes without having
to wait for a new release candidate. Though this might of course also contain some
unstable code.

In this case, instead of requiring a specific _version_ of Contao, a specific _branch_

> of Contao's public git repository will be required. Each minor version of Contao

Reviewer: ```suggestion
of Contao's public Git repository will be required. Each minor version of Contao
```

`4.9` version:

```json
{
    "name": "contao/managed-edition",
    "type": "project",
    "description": "Contao Open Source CMS",
    "require": {
        "php": "^7.1",
        "contao/calendar-bundle": "4.9.x-dev",
        "contao/comments-bundle": "4.9.x-dev",
        "contao/conflicts": "@dev",
        "contao/core-bundle": "4.9.x-dev",
        "contao/faq-bundle": "4.9.x-dev",
        "contao/installation-bundle": "4.9.x-dev",
        "contao/listing-bundle": "4.9.x-dev",
        "contao/manager-bundle": "4.9.x-dev",
        "contao/news-bundle": "4.9.x-dev",
        "contao/newsletter-bundle": "4.9.x-dev"
    },
    "conflict": {
        "contao-components/installer": "<1.3"
    },
    "extra": {
        "contao-component-dir": "assets",
        "symfony": {
            "require": "^4.2"
        }
    },
    "scripts": {
        "post-install-cmd": [
            "Contao\\ManagerBundle\\Composer\\ScriptHandler::initializeApplication"
        ],
        "post-update-cmd": [
            "Contao\\ManagerBundle\\Composer\\ScriptHandler::initializeApplication"
        ]
    }
}
```

Each time the packages are updated, the most recent code for this branch will be 

> pulled from Contao's public git repository.

Reviewer: ```suggestion
pulled from Contao's public Git repository.
```
